internal/guest/storage/scsi/scsi.go

```diff
@@ -50,6 +52,18 @@ func Mount(ctx context.Context, controller, lun uint8, target string, readonly b
 		trace.Int64Attribute("controller", int64(controller)),
 		trace.Int64Attribute("lun", int64(lun)))
 
+	if readonly {
+		// containers only have read-only layers so only enforce for them
+		var deviceHash string
+		if verityInfo != nil {
+			deviceHash = verityInfo.RootDigest
+		}
+		err = securityPolicy.EnforceDeviceMountPolicy(target, deviceHash)
+		if err != nil {
+			return errors.Wrapf(err, "won't mount scsi controller %d lun %d onto %s", controller, lun, target)
+		}
+	}
```

Comment:
now that I revisited this, I don't see the actual logic that creates the dm-verity target?

Reply:
ugh, great not sure what happened.

Reply:
Doing in a followup PR.